presto-main/src/main/java/com/facebook/presto/operator/scalar/CharacterStringCasts.java

```diff
@@ -0,0 +1,198 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.annotation.UsedByGeneratedCode;
+import com.facebook.presto.metadata.OperatorType;
+import com.facebook.presto.metadata.Signature;
+import com.facebook.presto.metadata.SqlScalarFunction;
+import com.facebook.presto.metadata.SqlScalarFunctionBuilder.SpecializeContext;
+import com.facebook.presto.spi.type.CharType;
+import com.facebook.presto.spi.type.Chars;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.VarcharType;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceUtf8;
+import io.airlift.slice.Slices;
+
+import static com.facebook.presto.metadata.FunctionKind.SCALAR;
+import static com.facebook.presto.metadata.OperatorType.CAST;
+import static com.facebook.presto.metadata.OperatorType.SATURATED_FLOOR_CAST;
+import static com.facebook.presto.metadata.Signature.withVariadicBound;
+import static com.facebook.presto.spi.type.Chars.trimSpacesAndTruncateToLength;
+import static com.facebook.presto.spi.type.StandardTypes.CHAR;
+import static com.facebook.presto.spi.type.StandardTypes.VARCHAR;
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+import static com.facebook.presto.spi.type.Varchars.isVarcharType;
+import static com.facebook.presto.spi.type.Varchars.truncateToLength;
+import static com.facebook.presto.util.Types.checkType;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.airlift.slice.SliceUtf8.countCodePoints;
+import static io.airlift.slice.SliceUtf8.offsetOfCodePoint;
+import static io.airlift.slice.Slices.EMPTY_SLICE;
+import static java.lang.Math.min;
+
+public final class CharacterStringCasts
+{
+    public static final SqlScalarFunction VARCHAR_TO_VARCHAR_CAST = truncateCast(VARCHAR, VARCHAR);
+    public static final SqlScalarFunction CHAR_TO_CHAR_CAST = truncateCast(CHAR, CHAR);
+    public static final SqlScalarFunction VARCHAR_TO_CHAR_CAST = trimCast(VARCHAR, CHAR);
+    public static final SqlScalarFunction CHAR_TO_VARCHAR_CAST = padSpacesCast(CHAR, VARCHAR);
+    public static final SqlScalarFunction VARCHAR_TO_CHAR_SATURATED_FLOOR_CAST = varcharToCharSaturatedFloorCast();
+
+    private CharacterStringCasts() {}
+
+    private static SqlScalarFunction truncateCast(String sourceType, String resultType)
+    {
+        Signature signature = getCastSignature(CAST, sourceType, resultType);
```

Comment:
Yes, it waits for #5377. It also requires https://github.com/Teradata/presto/pull/213 for literal parameters. However: https://github.com/Teradata/presto/pull/213 also waits for #5377